src/node/tendermint_internals.mli

```diff
@@ -0,0 +1,94 @@
+open Crypto
+
+type height = int64 [@@deriving yojson]
+
+type round = int [@@deriving yojson]
+(** At a specific (chain) height, Tendermint's consensus algorithm may run several rounds.*)
+
+(** Tendermint sometimes decides on a `Nil` value. *)
+type value =
+  | Block of Protocol.Block.t
+  | Nil
+[@@deriving yojson]
+
+val repr_of_value : value -> value
+(* TODO: ConsensusStep2 Tendermint uses repr of values instead of values. *)
+
+(* FIXME: ConsensusStep2 this is copied bad design. *)
+val produce_value : (State.t -> value) ref
+
+val is_valid : State.t -> value -> bool
+(** Ensures a value going through Tendermint is valid. TODO: ConsensusStep2 actually verify signatures, logging system. *)
+
+val on_block : nil_default:'a -> (Protocol.Block.t -> 'a) -> value -> 'a
+
+val block : Protocol.Block.t -> value
+
+val nil : value
+
+val update_value : value -> round -> value
+
+type consensus_step =
+  | Proposal
+  | Prevote
+  | Precommit
+      (** Tendermint's consensus goes through 3 steps at each round. Used inside a consensus instance in a node. *)
+
+val string_of_step : consensus_step -> string
+
+(** Tendermint's consensus step-communication with other nodes. Two first fields are height and
+    round of when the operation was sent. *)
+type sidechain_consensus_op =
+  | ProposalOP  of (height * round * value * round)
```

Comment:
Nitpick: `ProposalOP` expands to 3 words: `Proposal O* P*`. I'm guessing this is short for `ProposalOperation`, in which case it should be `ProposalOp`. Or just `ProposalOperation`.